cmsl1t/analyzers/BaseAnalyzer.py

```diff
@@ -88,4 +122,14 @@ def get_histogram_filename(self):
         return os.path.join(self.output_folder, output_file)
 
     def add_plotter(self, plotter):
+        """
+        Register a plotter with this analyzer, and set up it's outputs
+        """
+        file_format = self.config.try_get('output', 'plot_format', "png")
+        plotter.set_plot_output_cfg(self.plots_folder, file_format)
         self.all_plots.append(plotter)
+
+    def might_contain_histograms(self, filename):
+        this_file = "{}_histograms.root".format(self.name)
```

Comment:
Might be nicer if this and the `get_histogram_filename` method shared the file format string, maybe make it into a class variable and use in each method?